rpm/0072-sailfishos-gecko-Fix-audio-underruns-for-fullduplex-.patch

```diff
@@ -250,15 +256,19 @@ index 1cac5eadd45b..13a205e5383d 100644
 +      cubeb_fifo_read_done(&fdx->record, record_len);
 +      cubeb_fifo_write_done(&fdx->playback, playback_len);
 +
-+      // Drop the chunk of data if the latency is higher than 300 ms to keep
-+      // audio and video in sync. Maybe we need a better algorithm here.
++      // Drop the half of the data if the latency is higher than 300 ms to keep
++      // audio and video in sync. It's important to keep feeding the sink input,
++      // otherwise the latency calculator will get stuck.
 +      r = WRAP(pa_stream_get_latency)(stm->output_stream, &latency, NULL);
-+      if (r == 0 && latency > 300 * PA_USEC_PER_MSEC) {
-+        cubeb_fifo_read_done(&fdx->playback, playback_len);
-+      } else {
-+        // Write playback data.
-+        cubeb_fullduplex_play(stm);
++      LOGV("Latency: %lld", latency);
++      if (r == 0 && latency > 300 * PA_USEC_PER_MSEC && got > 1) {
++        size_t drop_len = MIN(playback_len / 2, cubeb_fifo_size(&fdx->playback));
```

Comment:
I can't claim to understand this code, but is it safe for `drop_len` to be something other than a multiple of `fdx->out_frame_size` here?

Reply:
Thank you for pointing me on that. I fixed this in the new version of the commit. Also, just in case, I added a code to handle the error value from the callback.

Reply:
Great, thank you!